test/observer.test.tsx

```diff
@@ -416,6 +416,64 @@ function runTestSuite(mode: "observer" | "useObserver") {
             expect(container.querySelector("span")!.innerHTML).toBe("1")
         })
     })
+
+    describe("error handling", () => {
+        test("errors should propagate", () => {
+            const x = mobx.observable.box(1)
+            const errorsSeen: any[] = []
+
+            class ErrorBoundary extends React.Component {
+                public static getDerivedStateFromError() {
+                    return { hasError: true }
+                }
+
+                public state = {
+                    hasError: false
+                }
+
+                public componentDidCatch(error: any, info: any) {
+                    errorsSeen.push("" + error)
+                }
+
+                public render() {
+                    if (this.state.hasError) {
+                        return <span>Saw error!</span>
+                    }
+                    return this.props.children
+                }
+            }
+
+            const C = obsComponent(() => {
+                if (x.get() === 42) {
+                    throw new Error("The meaning of life!")
+                }
+                return <span>{x.get()}</span>
+            })
+
+            // tslint:disable-next-line
+            const origErrorLogger = console.error
+            // tslint:disable-next-line
+            console.error = function() {
+                // supress printing warnings that React always prints in DEV, even with error boundaries...
+            }
+            try {
+                const rendered = render(
+                    <ErrorBoundary>
+                        <C />
+                    </ErrorBoundary>
+                )
+                expect(rendered.container.querySelector("span")!.innerHTML).toBe("1")
+                act(() => {
+                    x.set(42)
+                })
+                expect(errorsSeen).toEqual(["Error: The meaning of life!"])
```

Comment:
```suggestion
                expect(errorsSeen).toMatchSnapshot(mode)
```

Reply:
meh, it makes the test itself harder to read I think? (I usually favor `toMatchInlineSnapshot` over `toMatchSnapshot` for small stuff, but that wouldn't help much in this case)

Reply:
But if you prefer I'll replace it

Reply:
I tried inline snapshot, but in this case it doesn't work because the test suite is executed twice and it doesn't like it :)